src/hints/dict/felt252DictEntryInit.ts

```diff
@@ -0,0 +1,51 @@
+import { z } from 'zod';
+
+import { Felt } from 'primitives/felt';
+import { VirtualMachine } from 'vm/virtualMachine';
+
+import { resOperand, ResOperand } from 'hints/hintParamsSchema';
+import { HintName } from 'hints/hintName';
+
+/** Zod object to parse Felt252DictEntryInit hint */
+export const felt252DictEntryInitParser = z
+  .object({
+    Felt252DictEntryInit: z.object({
+      dict_ptr: resOperand,
+      key: resOperand,
+    }),
+  })
+  .transform(({ Felt252DictEntryInit: { dict_ptr, key } }) => ({
+    type: HintName.Felt252DictEntryInit,
+    dictPtr: dict_ptr,
+    key,
+  }));
+
+/**
+ * Felt252DictEntryInit hint
+ *
+ * Initialize the entry to be added to a given dictionnary.
+ */
+export type Felt252DictEntryInit = z.infer<typeof felt252DictEntryInitParser>;
+
+/**
+ * Get the dictionnary `dict` at `dictPtr`
+ * Get the `key` to be initialized
+ *
+ * Read the current value at `key` if any,
+ * Initialize `dict[key]` to 0 if undefined.
+ *
+ * Assert the value at `key` to the `dict` segment.
+ *
+ */
+export const felt252DictEntryInit = (
+  vm: VirtualMachine,
+  dictPtr: ResOperand,
+  key: ResOperand
+) => {
+  const address = vm.getPointer(...vm.extractBuffer(dictPtr));
+  const keyValue = vm.getResOperandValue(key).toString();
+  const dict = vm.getDict(address);
+  const prevValue = dict.get(keyValue) || new Felt(0n);
```

Comment:
Setting the previous value to `0` when initializing a key is the convention for Cairo dicts afaik

Taken from the [Cairo book](https://book.cairo-lang.org/ch03-02-dictionaries.html):
> For each time we interact with a Felt252Dict<T>, a new Entry<T> will be registered:
>
>    A `get` would register an entry where there is no change in state, and previous and new values are stored with the same value.
>    An `insert` would register a new `Entry<T>` where the `new_value` would be the element being inserted, and the `previous_value` the last element inserted before this. In case it is the first entry for a certain key, then the previous value will be zero.

[Rust implementation](https://github.com/lambdaclass/cairo-vm/blob/37ea72977dccbc2b90b8b7534c1edabd2e2fef79/vm/src/hint_processor/cairo_1_hint_processor/hint_processor.rs#L783-L785), [Go implementation](https://github.com/NethermindEth/cairo-vm-go/blob/be3418769457710ad529a5bb59c461b720cce380/pkg/hintrunner/core/hint.go#L826-L830)

Reply:
this doesn't surprise me for cairo1 i'm a bit more surprised for cairo0 (but perhaps i'm out of topic and this is only for cairo1 stuff? in which case you can disregard my comment)

```
func default_dict_new(default_value: felt) -> (res: DictAccess*) {
    %{
        if '__dict_manager' not in globals():
            from starkware.cairo.common.dict import DictManager
            __dict_manager = DictManager()

        memory[ap] = __dict_manager.new_default_dict(segments, ids.default_value)
    %}
    ap += 1;
    return (res=cast([ap - 1], DictAccess*));
}
```

The dict manager registers what the default value of a default_dict is - and it's not necessarily zero.

see: https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/dict.py#L44-L55

```
    def new_default_dict(self, segments, default_value, temp_segment: bool = False):
        """
        Creates a new Cairo default dictionary.
        """
        base = segments.add_temp_segment() if temp_segment else segments.add()
        assert base.segment_index not in self.trackers
        self.trackers[base.segment_index] = DictTracker(
            data=defaultdict(lambda: default_value),
            current_ptr=base,
        )
        return base
```

data is implemented as a defaultdict, where if a value is not initialized in the dict, the lambda function will be executed and will return the default value.

Reply:
ok great! thank you